developer/src/kmcmplib/src/CompilerInterfaces.cpp

```diff
@@ -133,105 +132,70 @@ EXTERN bool kmcmp_CompileKeyboardFile(char* pszInfile,
     SetError(CERR_CannotCreateTempfile);
   }
 
-  fp_out = Open_File(pszOutfile, "wb");
-
-  if (fp_out == NULL) {
-    SetError(CERR_CannotCreateOutfile);
-  }
-
-  FILE_KEYBOARD fk;
   kmcmp::CodeConstants = new kmcmp::NamedCodeConstants;
-  bool result = CompileKeyboardHandle(fp_in, &fk);
-  if(result) {
-    KMX_DWORD msg;
-    if ((msg = WriteCompiledKeyboard(&fk, fp_out)) != CERR_None) {
-      result = FALSE;
-      AddCompileError(msg);
-    }
-  } else {
-    AddCompileError(CERR_InvalidValue);
-  }
+  bool result = CompileKeyboardHandle(fp_in, static_cast<PFILE_KEYBOARD>(pfkBuffer));
```

Comment:
We don't need `AddCompileError` if `CompileKeyboardHandle` failed?

Reply:
`CompileKeyboardHandle` already calls `AddCompileError` if it fails, with more detail.